caffe/src/caffe/layers/molgrid_data_layer.cpp

```diff
@@ -329,7 +335,22 @@ void MolGridDataLayer<Dtype>::DataLayerSetUp(const vector<Blob<Dtype>*>& bottom,
       while (getline(infile, line))
       {
         example ex(line, hasaffinity, hasrmsd);
-        data2.add(ex);
+        if (paired)
```

Comment:
This code is duplicative.  I would keep the add method, but pass it the paired/balanced bools (or make them members of the data structure and set them in the constructor).